src/screens/ContactScreen/index.js

```diff
@@ -404,16 +470,28 @@ const ContactScreen = ({navigation}) => {
         />
       )}
 
-      {isSearchMode && (
-        <SearchRecyclerView
-          text={debouncedText}
-          followed={followed}
-          selectedUsers={selectedUsers}
-          usernames={usernames}
-          setLoading={setIsLoadingSearchResult}
-          onHandleSelected={(value) => handleSelected(value)}
+      {isRecyclerViewShownSearch && usersSearch.length > 0 && (
+        <RecyclerListView
+          style={styles.recyclerview}
+          layoutProvider={layoutProviderSearch}
+          dataProvider={dataProviderSearch}
+          extendedState={{
+            followed
+          }}
+          rowRenderer={rowRendererSearch}
+          scrollViewProps={{
+            refreshControl: (
+              <RefreshControl
+                tintColor={COLORS.white}
+                refreshing={refreshing}
+                onRefresh={onRefresh}
+              />
+            )
+          }}
         />
       )}
+
+      {isLoadingSearchResult && <LoadingWithoutModal />}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The RecyclerView for the search results has been added. It uses the state variables and functions related to search, separating its logic from the main user list. This is a good practice as it makes the code more maintainable. However, the `isLoadingSearchResult` state variable is used to show a loading spinner, but it's not clear where this variable is defined or updated. Please ensure that it's properly managed.


<!-- This is an auto-generated comment by OSS CodeRabbit -->